SonarQube.Common/Utilities.cs

```diff
@@ -64,6 +65,54 @@ public static bool Retry(int timeoutInMilliseconds, int pauseBetweenTriesInMilli
             return succeeded;
         }
 
+        /// <summary>
+        /// Ensures that the specified directory exists
+        /// </summary>
+        public static void EnsureDirectoryExists(string directory, ILogger logger)
```

Comment:
New method for creating new, empty directories with logging. I've refactored the existing product code that creates directories to use either this method or the one below, as appropriate.